packages/vuetify/src/composables/items.ts

```diff
@@ -0,0 +1,77 @@
+// Utilities
+import { computed } from 'vue'
+import { getObjectValueByPath, getPropertyFromItem, propsFactory } from '@/util'
+
+// Types
+import type { PropType } from 'vue'
+
+export interface InternalItem {
+  [key: string]: any
+  title: string
+  value: any
+  children?: InternalItem[]
+}
+
+export interface ItemProps {
+  items: (string | Partial<InternalItem>)[]
+  itemTitle: string
+  itemValue: any
+  itemChildren: string
+  itemProps: (item: any) => Partial<InternalItem>
+}
+
+// Composables
+export const makeItemsProps = propsFactory({
+  items: {
+    type: Array as PropType<ItemProps['items']>,
+    default: () => ([]),
+  },
+  itemTitle: {
+    type: String,
```

Comment:
`getPropertyFromItem` supports `SelectItemKey`

Reply:
Would the same logic apply to **itemValue**? Should **itemChildren** be updated in some form since it uses **getObjectValueByPath**?

Reply:
- yes
- yeah we were using `getPropertyFromItem` before but it returns the item if the property is undefined